manager/integration/tests/test_basic.py

```diff
@@ -2051,3 +2053,30 @@ def test_expansion_canceling(clients, core_api, volume_name, pod):  # NOQA
     delete_and_wait_pod(core_api, pod_name)
     delete_and_wait_pvc(core_api, expansion_pvc_name)
     delete_and_wait_pv(core_api, expansion_pv_name)
+
+
+@pytest.mark.coretest  # NOQA
+@pytest.mark.skip(reason="TODO")
+def test_running_volume_with_scheduling_failure():
+    """
+    Test if the running volume still work fine
+    when there is a scheduling failed replica
+
+    1. Disable the soft anti-affinity setting.
+    2. Create a volume, then create the corresponding PV, PVC and Pod.
+    3. Wait for the pod running and the volume healthy.
+    4. Write data to the pod volume and get the md5sum.
+    5. Disable a node scheduling.
+    6. Crash the replica on the scheduling disabled node for the volume.
+    7. Wait for the new replica created.
+    8. Verify the volume is Degraded and fails to scheduled the replica.
+    9. Verify the `volume.ready == True`.
+    10. Delete the pod and wait for the volume detached.
+    11. Verify the failed replica is removed and
+        the volume contains healthy replicas only.
+    12. Verify the `volume.ready == True`.
```

Comment:
The volume will be in this state isnt it error - `Scheduling Failure Replica Scheduling Failure` and here volume read = No ?

Reply:
No. 
`Scheduling Failure` is caused by scheduling failed replica. But this failed replica will be cleaned up during the volume detachment (step 10) then the error will disappear. 
Otherwise, the error will prevent the volume from being reattached/reused by the workloads and it would become a bug.